docs/docs/ecosystem/helicone.md

```diff
@@ -0,0 +1,67 @@
+# Helicone
+
+This page covers how to use the [Helicone](https://helicone.ai) within LangChain.
+
+## What is Helicone?
+
+Helicone is an [open source](https://github.com/Helicone/helicone) observability platform that proxies your OpenAI traffic and provides you key insights into your spend, latency and usage.
+
+![Helicone](../_static/HeliconeDashboard.png)
+
+## Quick start
+
+With your LangChain environment you can just add the following parameter.
+
+```typescript
+const model = new OpenAI(
+  {},
+  {
+    basePath: "https://oai.hconeai.com/v1",
+  }
+);
+const res = await model.call("What is a helicone?");
+```
+
+Now head over to [helicone.ai](https://helicone.ai/onboarding?step=2) to create your account, and add your OpenAI API key within our dashboard to view your logs.
+
+![Helicone](../_static/HeliconeKeys.png)
+
+## How to enable Helicone caching
+
+```typescript
+const model = new OpenAI(
+  {},
+  {
+    basePath: "https://oai.hconeai.com/v1",
+    baseOptions: {
+      headers: {
+        "Helicone-Cache-Enabled": "true",
+      },
+    },
+  }
+);
+const res = await model.call("What is a helicone?");
+```
+
+[Helicone caching docs](https://docs.helicone.ai/advanced-usage/caching)
+
+## How to use Helicone custom properties
+
+```typescript
+const model = new OpenAI(
+  {},
+  {
+    basePath: "https://oai.hconeai.com/v1",
+    baseOptions: {
+      headers: {
+        "Helicone-Property-Session": "24",
```

Comment:
Could be useful to provide some utility function for constructing these Helicone headers, e.g. this could be something like:

```ts
const model = new OpenAI(
    {},
    {
       ...HeliconeOptions({
          propertySession: "24",
          propertyConversation: "support_issue_2",
          propertyApp: "mobile",
          cacheEnabled: false
       }),
       otherAxiosConfigurationOption: "value"
    }
)
```

Reply:
this is good idea, will add issue